src/main/java/uk/gov/hmcts/darts/common/config/security/SecurityConfig.java

```diff
@@ -138,4 +150,54 @@ protected void doFilterInternal(HttpServletRequest request, HttpServletResponse
         }
     }
 
-}
+    @RequiredArgsConstructor
+    private final class InactiveUserAuthorisationCheck extends OncePerRequestFilter {
+
+        private final ObjectMapper mapper;
+
+        @Override
+        protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain filterChain)
+            throws ServletException, IOException {
+            Jwt jwt = null;
+            try {
+                var jwtDecoder = NimbusJwtDecoder.withJwkSetUri(externalAuthProviderConfigurationProperties.getJwkSetUri())
+                    .jwsAlgorithm(SignatureAlgorithm.RS256)
+                    .build();
+
+                String authHeader = request.getHeader("Authorization");
+
+                OAuth2TokenValidator<Jwt> jwtValidator = JwtValidators.createDefaultWithIssuer(externalAuthConfigurationProperties.getIssuerUri());
+                jwtDecoder.setJwtValidator(jwtValidator);
+                jwt = jwtDecoder.decode(authHeader.replace(TOKEN_BEARER_PREFIX, "").trim());
+            } catch (Exception exception) {
+                log.error("Problem decoding the token", exception);
+            }
+
+            // if we cant determine the jwt then proxy and let the nimbus library handle the 401
+            if (jwt == null) {
+                filterChain.doFilter(request, response);
+            } else {
+                try {
+                    UserAccountEntity userAccountEntity = userIdentity.getUserAccount(jwt);
+                    if (!userAccountEntity.isActive()) {
+                        writeError(response);
+                    } else {
+                        filterChain.doFilter(request, response);
+                    }
+                } catch (Exception exception) {
+                    log.error("User is invalid", exception);
+                    writeError(response);
+                }
+            }
+        }
+
+        private void writeError(HttpServletResponse response) {
+            try {
+                DartsApiTrait.writeErrorResponse(response, mapper);
+            } catch (IOException ex) {
+                log.error("Problem parsing the problem", ex);
+                response.setStatus(HttpStatus.FORBIDDEN.value());
+            }
+        }
```

Comment:
Not sure I understand this bit, could you explain? Can't we throw a DartsApiException here (like we do elsewhere in our application) or are the exception handling mechanisms not in play at this early stage of processing the request?

Reply:
When we return the 403 forbidden I wanted to throw a forbidden message that was consistent with other parts of the system. See getUserAccount.getUserAccount. 

I would have thrown an exception but the Spring traits are not active at this point, so have to write the Spring response myself